### ISSUE TITLE
Not generating graph

### ISSUE THREAD

**Issue by adbrei** [varunshenoy/GraphGPT#19]
Hello,

I installed everything, and the UI in the browser opens as expected. However, when I type my prompt and click 'Generate' nothing seems to happen. I've tried in different locations with different WiFi, so I don't think it's the WiFi connection. I've left it open for about fifteen minutes. Does anyone have any thoughts/recommendations? Thanks!

**Comment by iAmmarTahir** [comment]
Have you added your OPENAI_API_KEY?

**Comment by adbrei** [comment]
Yes, I added my key to App.js, line 5:
`const OPENAI_API_KEY = "MY_OPENAI_API_KEY"`

**Comment by lodm94** [comment]
How did you manage to launch te UI locally?
I did the following:
- apt update && apt install npm
- npm install react-graph-vis
- copied the API kei
- npm run start

But it fails like i reported [here](https://github.com/varunshenoy/GraphGPT/issues/18).
Maybe its a host problem? I launch this on windows using ubuntu WSL2

**Comment by iAmmarTahir** [comment]
I have these versions:
Node: v14.17.4
NPM: 7.9.1
and everything works fine for me. 
Can you guys try this

**Comment by lodm94** [comment]
yeah i will. but which is your host device? Ubuntu laptop? Server?

**Comment by iAmmarTahir** [comment]
> yeah i will. but which is your host device? Ubuntu laptop? Server?

I'm on Macbook M1

**Comment by lodm94** [comment]
I will try on the same machine. But do you think these steps are ok or am i missing something?


> * apt update && apt install npm
> * npm install react-graph-vis
> * copied the API kei
> * npm run start

**Comment by iAmmarTahir** [comment]
You need to run `npm install`

**Comment by adbrei** [comment]
@lodm94 I had those same errors until I ran `npm install` Now, the UI opens in the browser; however, nothing happens when I click 'generate' (I've left it open for an hour and a half).
<img width="682" alt="Screenshot 2023-02-03 at 8 03 19 AM" src="https://user-images.githubusercontent.com/60265275/216611032-6d2a4eb2-3f0f-477c-bddb-a57bed2f0a9b.png">
<img width="1624" alt="Screenshot 2023-02-03 at 8 05 14 AM" src="https://user-images.githubusercontent.com/60265275/216611040-d108e7a9-2b52-461b-af25-1d9932f767b5.png">

**Comment by lodm94** [comment]
Thanks all for the support, i figured it out.
In Ubuntu 20.04 i needed to do the following:

- `apt update && apt install npm`
- `npm i -g npm@latest` 
- `curl -fsSL https://deb.nodesource.com/setup_18.x | sudo -E bash - && sudo apt-get install -y  nodejs` 
- `npm install`
- `npm start`

Then i navigate to localhost:3000 and the UI is up&running.
This is the only way to make it work; i tried on a Ubuntu 20.04 workstation and also on Windows 11 workstation using WSL2.
Do you think it is appropriate to include these steps in the instructions?

**Comment by iAmmarTahir** [comment]
I have created `https://github.com/iAmmarTahir/KnowledgeGraphGPT`. That solves most of the issues: 
- Graph generation for longer texts
- Exporting them
- Coloring them
- Iteratively adding new information, etc.

**Comment by ramonwenger** [comment]
@iAmmarTahir I'm sure you didn't mean any harm by it, but I would find it much more productive for you to contribute those changes to this repo by making a PR instead of creating your own and advertising it in the issues and thus potentially splitting up the potential user base of either. The repo has already a few contributions, and @varunshenoy is quite quick in replying / merging.

**Comment by iAmmarTahir** [comment]
> @iAmmarTahir I'm sure you didn't mean any harm by it, but I would find it much more productive for you to contribute those changes to this repo by making a PR instead of creating your own and advertising it in the issues and thus potentially splitting up the potential user base of either. The repo has already a few contributions, and @varunshenoy is quite quick in replying / merging.

Yeah, I think you're right about it.

**Comment by cozy-17** [comment]
hy @lodm94 hou u fixed that i have windows u know how to solve that in windows?
